lib/run-script-pkg.js

```diff
@@ -41,7 +47,15 @@ const runScriptPkg = async options => {
   )
     cmd = defaultGypInstallScript
   else if (event === 'start' && await isServerPackage(path))
-    cmd = 'node server.js' + args.map(a => ` ${JSON.stringify(a)}`).join('')
+    // `node` probably means `C:\Program Files\nodejs\node.exe`.
+    // But it can also point to `.\node_modules\.bin\node.cmd` if you install a
+    // Node.js version using `npm i node`.
+    // So it might point to an executable, or a batch file. They require
+    // different escaping. Also, `puka` only supports batch-style escaping.
+    // The solution is to always use a batch file on Windows.
```

Comment:
You’re right, this won’t be using the batch file wrapper. `node` is most likely an executable, so the escaping would be wrong (except in cases where people have installed `node` using `npm i node`). Dang, Windows escaping sucks :( See this comment for more details: https://github.com/npm/run-script/pull/31#issuecomment-843562437

Running `node` for a script like in your example feels like a very common use case, so it would be nice if that worked nicely.

Reply:
The more I think about this, the more I realize something:

- I don’t use `-- extra args` with `npm run` very much, so I don’t care super much about that. It’s a bit of an impossible task to append extra args to an unknown shell string.

- I do use `npx` a lot and would like it to be intuitive. I think implementing it in terms of `npm run` feels pretty hacky – it’s like creating a string and using `eval()` in JavaScript for something that can be done just fine in the language.